### ISSUE TITLE
Throwing in riot.observable event handler results in handler never being called again

### ISSUE THREAD

**Issue by josschne** [riot/observable#3]
Moved from https://github.com/riot/riot/issues/806

When a riot.observable event handler throws, the handler's busy bit is set and can never be cleared. This results in the handler never being called again for subsequent event triggers.

Example at: http://jsfiddle.net/josschne/3se9L3sv/

Suggested fix:

```
if (!fn.busy) {
    fn.busy = 1
    try {
      fn.apply(el, fn.typed ? [name].concat(args) : args)
    } finally {
      if (fns[i] !== fn) { i-- }
      fn.busy = 0
    }
  }
}
```

**Comment by GianlucaGuarini** [comment]
I had to revert this change this behavior will be removed from the next riot release. Hiding the errors internally it's not a good idea

**Comment by yuretz** [comment]
Have I understood it correctly, that starting from riot 2.3.13 any handler that throws exception once will remain busy forever, like it was before bed80732108c6ddfacabb7d55f600fd2e84f0524 ? Are you going to address this issue somehow in the future releases?

**Comment by GianlucaGuarini** [comment]
> handler that throws exception once will remain busy forever..

yes! the solution is: **you must not have javascript errors thrown in your apps** it's fool to hide errors behind core modules.. eventually you can handle the errors outside in `try/catch` blocks

**Comment by yuretz** [comment]
Well, exceptions sometimes _do get thrown_, and not always at our will or expectations. I absolutely agree, that they should never be hidden.So it's only a question about how and where to handle them. 

I believe it's totally OK if you leave the user code responsible for that, but it looks very wrong when the state of observable itself becomes _irreversibly inconsistent_ just because some external code that you called has thrown an exception, and you decided not to handle it yourself but pass the trouble further to the caller code.

IMO, you either have to get rid of the `busy` flag, or, at least, reset it in the `finally` block. Otherwise, what you are doing is simply forcing the user of your library to always have his exception handling in the event handler, just not to break the observable instance. And since you've just declared that it's not observable's responsibility to catch those damn exceptions, then it should certainly be up to the user to decide where he has his `try..catch`es, your code shouldn't break because of it.

**Comment by GianlucaGuarini** [comment]
A solution can be eventually to reset the busy flag and then throw the error. I will think about it for the next `riot-observable` release

**Comment by GianlucaGuarini** [comment]
this issue was solved in the v 3.0.0-alpha.1 the busy flag will be removed
